assembly/src/parsers/blocks.rs

```diff
@@ -35,6 +37,41 @@ pub fn parse_code_blocks(
     }
 }
 
+pub fn parse_proc_blocks(
+    tokens: &mut TokenStream,
+    proc_map: &BTreeMap<String, CodeBlock>,
+    proc_locals: u32,
+) -> Result<CodeBlock, AssemblyError> {
+    // parse the procedure body
+    let body = parse_code_blocks(tokens, proc_map, proc_locals)?;
+
+    if proc_locals == 0 {
+        // if no allocation of locals is required, return the procedure body
+        return Ok(body);
+    }
+
+    let mut blocks = Vec::new();
+    let locals_felt = Felt::new(proc_locals as u64);
+
+    // allocate procedure locals before the procedure body
+    let alloc_ops = vec![Operation::Push(locals_felt), Operation::FmpUpdate];
+    blocks.push(CodeBlock::new_span(alloc_ops));
+
+    // add the procedure body code block
+    blocks.push(body);
+
+    // deallocate procedure locals after the procedure body
+    let dealloc_ops = vec![
+        Operation::Push(locals_felt),
+        Operation::Neg,
```

Comment:
We could probably do this as a single operation: `Operation::Push(-locals_felt)`

Reply:
Yes, fixed!